pkg/metrics/leaderelection.go

```diff
@@ -0,0 +1,41 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/tools/leaderelection"
+)
+
+// This file is copied and adapted from k8s.io/component-base/metrics/prometheus/clientgo/leaderelection
+// which registers metrics to the k8s legacy Registry. We require very
+// similar functionality, but must register metrics to a different Registry.
+
+var (
+	leaderGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "leader_election_master_status",
```

Comment:
What means master status?

Reply:
It means whether this instance is master or backup, as described in the help field. The name is just grabbed from k8s.io/component-base which is also used by kube-controller-manager.